src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -402,6 +386,243 @@ public new string Name
+        ** general then IsAlive or IsBackground.
+        =========================================================================*/
+        public ThreadState ThreadState
+        {
+            get { return (ThreadState)GetThreadStateNative(); }
+        }
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private extern int GetThreadStateNative();
+
+        public ApartmentState GetApartmentState()
+        {
+#if FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            return (ApartmentState)GetApartmentStateNative();
+#else // !FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            Debug.Assert(false); // the Thread class in CoreFX should have handled this case
+            return ApartmentState.MTA;
+#endif // FEATURE_COMINTEROP_APARTMENT_SUPPORT
+        }
+
+        /*=========================================================================
+        ** An unstarted thread can be marked to indicate that it will host a
+        ** single-threaded or multi-threaded apartment.
+        =========================================================================*/
+        public bool TrySetApartmentStateUnchecked(ApartmentState state)
+        {
+#if FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            return SetApartmentStateHelper(state, false);
+#else // !FEATURE_COMINTEROP_APARTMENT_SUPPORT
+            Debug.Assert(false); // the Thread class in CoreFX should have handled this case
+            return false;
+#endif // FEATURE_COMINTEROP_APARTMENT_SUPPORT
+        }
+
+#if FEATURE_COMINTEROP_APARTMENT_SUPPORT
+        internal bool SetApartmentStateHelper(ApartmentState state, bool fireMDAOnMismatch)
+        {
+            ApartmentState retState = (ApartmentState)SetApartmentStateNative((int)state, fireMDAOnMismatch);
+
+            // Special case where we pass in Unknown and get back MTA.
+            //  Once we CoUninitialize the thread, the OS will still
+            //  report the thread as implicitly in the MTA if any
+            //  other thread in the process is CoInitialized.
+            if ((state == System.Threading.ApartmentState.Unknown) && (retState == System.Threading.ApartmentState.MTA))
+                return true;
+
+            if (retState != state)
+                return false;
+
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal extern int GetApartmentStateNative();
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal extern int SetApartmentStateNative(int state, bool fireMDAOnMismatch);
+#endif // FEATURE_COMINTEROP_APARTMENT_SUPPORT
+
+#if FEATURE_COMINTEROP
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        public extern void DisableComObjectEagerCleanup();
+#else // !FEATURE_COMINTEROP
+        public void DisableComObjectEagerCleanup()
+        {
+            Debug.Assert(false); // the Thread class in CoreFX should have handled this case
```

Comment:
The Thread class in corefx is going away.

Reply:
I removed the `Assert`. At first sight it looks like it should be `throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);`, but CoreFX silently ignored this so I am preserving the behaviour.

Reply:
This is intentionally no-op.

We try to make COM-specific APIs to have reasonable behavior when COM is not present. For example, `Marshal.IsComObject` returns `false` on Unix - it does not throw PNSE.